scripts/scil_validate_bids.py

```diff
@@ -188,106 +141,112 @@ def get_data(nSub, dwi, t1s, fs, associations, default_readout, clean):
-            topup = topup_fmap
+
+    if len(dwis) == 2:
+        dwi_path[1] = dwis[1].path
+        bvec_path[1] = layout.get_bvec(dwis[1].path)
+        bval_path[1] = layout.get_bval(dwis[1].path)
+        if 'direction' in dwis[1].entities:
+            PE[1] = conversion[dwis[1].entities['direction']]
+        elif 'PhaseEncodingDirection' in dwis[1].entities:
+            PE[1] = conversion[dwis[1].entities['PhaseEncodingDirection']]
+
+    curr_dwi = dwis[0]
+    dwi_path[0] = curr_dwi.path
+    bvec_path[0] = layout.get_bvec(curr_dwi.path)
+    bval_path[0] = layout.get_bval(curr_dwi.path)
+
+    if 'TotalReadoutTime' in curr_dwi.entities:
+        totalreadout = curr_dwi.entities['TotalReadoutTime']
+
+    if 'session' in curr_dwi.entities:
+        nSess = curr_dwi.entities['session']
+
+    if 'run' in curr_dwi.entities:
+        nRun = curr_dwi.entities['run']
+
+    IntendedForPath = os.path.sep.join(curr_dwi.relpath.split(os.path.sep)[1:])
+    if 'TotalReadoutTime' in curr_dwi.entities:
+        related_files = layout.get(part="mag",
+                                   IntendedFor=IntendedForPath,
+                                   regex_search=True,
+                                   TotalReadoutTime=totalreadout) +\
+                        layout.get(part=Query.NONE,
+                                   IntendedFor=IntendedForPath,
+                                   regex_search=True,
+                                   TotalReadoutTime=totalreadout)
+    else:
+        related_files = layout.get(part="mag",
+                                   IntendedFor=IntendedForPath,
+                                   regex_search=True) +\
+                        layout.get(part=Query.NONE,
+                                   IntendedFor=IntendedForPath,
+                                   regex_search=True)
+
+    direction = False
+    if 'direction' in curr_dwi.entities:
+        dwi_direction = curr_dwi.entities['direction']
+        direction = True
+    elif 'PhaseEncodingDirection' in curr_dwi.entities:
+        dwi_direction = curr_dwi.entities['PhaseEncodingDirection']
+
+    PE[0] = conversion[dwi_direction]
+    if len(related_files) == 1 and related_files[0].entities['suffix'] == 'epi' and len(dwis) == 1:
+        # Usual use case - 1 DWI + 1 fmap
+        if direction:
+            if dwi_direction[::-1] == related_files[0].entities['direction']:
+                topup_suffix['epi'][1] = related_files[0].path
+                PE[1] = conversion[related_files[0].entities['direction']]
+        elif 'PhaseEncodingDirection' in curr_dwi.entities:
+            PE[1] = conversion[related_files[0].entities['PhaseEncodingDirection']]
+            if dwi_direction == get_opposite_phase_encoding_direction(related_files[0].entities['PhaseEncodingDirection']):
+                topup_suffix['epi'][1] = related_files[0].path
+
+    elif len(related_files) >= 2:
+        for curr_related in related_files:
+            if direction:
```

Comment:
Can you use some intermediary variables to avoid this if/else ? The same could be done with the if/else at line 205